merkle/merkle_path.go

```diff
@@ -46,6 +47,83 @@ func CalcInclusionProofNodeAddresses(treeSize, index int64, maxBitLen int) ([]st
 	return proof, nil
 }
 
+// CalcConsistencyProofNodeAddresses returns the tree node IDs needed to
+// build a consistency proof between two specified tree sizes. The maxBitLen parameter
+// is copied into all the returned nodeIDs. The caller is responsible for checking that
+// the input tree sizes correspond to valid tree heads. All returned NodeIDs are tree
+// coordinates within the new tree. It is assumed that they will be fetched from storage
+// at a revision corresponding to the STH associated with the treeSize parameter.
+func CalcConsistencyProofNodeAddresses(previousTreeSize, treeSize int64, maxBitLen int) ([]storage.NodeID, error) {
+	if previousTreeSize > treeSize || previousTreeSize < 1 || treeSize < 1 || maxBitLen <= 0 {
+		return []storage.NodeID{}, fmt.Errorf("invalid params prior: %d treesize: %d, bitlen:%d", previousTreeSize, treeSize, maxBitLen)
+	}
+
+	return snapshotConsistency(previousTreeSize, treeSize, maxBitLen), nil
+}
+
+// snapshotConsistency does the calculation of consistency proof node addresses between
+// two snapshots. Based on the C++ code used by CT but adjusted to fit our situation.
+// In particular the code does not need to handle the case where overwritten node hashes
+// must be recursively computed because we have versioned nodes.
+func snapshotConsistency(snapshot1, snapshot2 int64, maxBitLen int) []storage.NodeID {
+	var proof []storage.NodeID
+
+	level := 0
+	node := snapshot1 - 1
+
+	// Compute the (compressed) path to the root of snapshot2.
+	// Everything left of 'node' is equal in both trees; no need to record.
+	for (node & 1) != 0 {
+		node /= 2
```

Comment:
Done. Thought about having parent and sibling funcs like C++ but seemed overkill.